india_compliance/gst_india/utils/transaction_data.py

```diff
@@ -396,15 +403,50 @@ def sanitize_value(
         min_length=3,
         max_length=100,
         truncate=True,
+        throw=False,
     ):
+        """
+        Sanitize a value for use in a GST document.
+        @param value: The value to be sanitized. Can be one of the following:
+            - A string
+            - A dict with a key matching the fieldname and a value
+            e.g. {"address_title": "Test Address"}, {"address_title": <dict> like object}
+        """
+
+        def _throw(error):
+            if throw:
+                frappe.throw(error, title=_("Missing Data"))
+
+        key = value
+        if isinstance(value, dict):
+            key = list(value.keys())[0]
+            value = value[key]
+
+            if isinstance(value, dict):
+                value = value.get(key)
+
         if not value or len(value) < min_length:
-            return
+            return _throw(
+                _("{0} must be at least {1} characters long").format(key, min_length),
+            )
+
+        if isinstance(value, str) and not value.isascii():
+            return _throw(
+                _(
+                    "Unicode characters are not allowed. Please edit details in English only."
+                )
+            )
 
         if regex:
             value = re.sub(REGEX_MAP[regex], "", value)
 
+        if not value:
+            return _throw(_("{0} is mandatory.").format(key))
```

Comment:
You are already checking this in 428. Duplication is not needed here.

Reply:
This validation is for value after regex is applied.